### PR TITLE
Datadir

### PR DESCRIPTION
**Pull request by ericgaspar**
- Set new datadir `/home/yunohost.transmission` -> `/home/yunohost.app/transmission`
- Add --purge option
but this PR don't manage the new data path for the already installed Transmission app...

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -45,6 +46,12 @@ fi
 # Add peer_port also on UDP.
 ynh_exec_warn_less yunohost firewall allow UDP $peer_port
 
+if [ -z "$datadir" ]; then
+	datadir=/home/yunohost.app/$app
+	ynh_app_setting_set --app=$app --key=datadir --value=$datadir
+	cp -a "/home/yunohost.transmission" "$datadir"
```
Comment (alexAubin) [inline scripts/upgrade:52]:
```suggestion
	mv "/home/yunohost.transmission" "$datadir"
```

I think I'd rather do a move ? Any particular reason to use a cp ?

File: scripts/upgrade [hunk 2]
```diff
@@ -90,16 +97,14 @@ ynh_install_app_dependencies $pkg_dependencies
 # CREATE DIRECTORIES
 #=================================================
 
-mkdir -p /home/yunohost.transmission/{progress,completed,watched}
+datadir=/home/yunohost.app/$app
```
Comment (alexAubin) [inline scripts/upgrade:100]:
```suggestion
```

That should not be needed considering the var is either fetched from the setting or auto-defined in the migration snippet earlier

**Review by alexAubin — APPROVED** [review]
LGTM apart from minor comments :+1:

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!tastemetitus

**Comment by ericgaspar** [comment]
!testme

**Pull request closed without merging**